tasks/main.yml

```diff
@@ -23,11 +23,26 @@
   when:
     - not network_packages is subset(ansible_facts.packages.keys())
 
-- name: Enable network service
+- name: Enable and start NetworkManager
   service:
       name: "{{ network_service_name }}"
       state: started
       enabled: yes
+  when:
+    - network_provider == "nm"
+
+- name: Enable network service
+  service:
+      name: "{{ network_service_name }}"
+      enabled: yes
+  when:
+    -  network_provider == "initscripts"
```

Comment:
When will it be actually started?

Reply:
I mean, don't you still need to start it to apply the new settings?

Reply:
No, the new settings are applied by the module (it issues `ifup` directly). The service only ensures that the profiles on disk are applied when the system starts (or when the service is started).